ports/posix/src/iot_threads_posix.c

```diff
@@ -200,6 +197,34 @@ bool Iot_CreateDetachedThread( IotThreadRoutine_t threadRoutine,
         IOT_SET_AND_GOTO_CLEANUP( false );
     }
 
+    if( stackSize != 0 )
```

Comment:
Could you use the constants `IOT_THREAD_DEFAULT_PRIORITY` and `IOT_THREAD_DEFAULT_STACK_SIZE` in iot_platform_types.h instead of hardcoded 0?  Thanks.

Reply:
Making that change would essentially cancel out these changes. We set `IOT_THREAD_DEFAULT_STACK_SIZE` & `IOT_THREAD_DEFAULT_PRIORITY` to the default value we want to be used to create threads by the SDK, such as for thread pools. The SDK internally will use this value when calling `Iot_CreateDetachedThread()`, if inside this function we are checking for this value and ignoring it if they are the same these would never be set as they will always match.

The only reason for checking for 0 is to see if the value of `IOT_THREAD_DEFAULT_STACK_SIZE` & `IOT_THREAD_DEFAULT_PRIORITY` were set in the configuration. If they were never set then these two macros will default to a value of 0, therefore the threads stack size and priority shouldn’t be set so that the system’s defaults are used.

Reply:
Understood, thank you.  I may change the priority comparison to `priority != -1` just because `0` may be a valid priority and this code would prevent that priority from being used.  Nothing further that you need to do though, just wanted to let you know.